### ISSUE TITLE
train yolov5 and optimize with Neutrino

### ISSUE THREAD

**Issue by sudhakar-sah** [Deeplite/deeplite-torch-zoo#37]
- [ ] train yolov5 and achieve SOTA

- [ ] Optimize with Neutrino 

- [ ] Update the performance report with yolov5 (VOC)

**Comment by wernerchaodeeplite** [comment]
Dataloader loading more data in DGX, 85K instead of the original 16K training set. Will take the following action:
1. Kill current training job, and debug data loader
2. Review yolo5 paper for proper training parameters, i.e. batch size and loss function
3. Train from scratch

**Comment by wernerchaodeeplite** [comment]
Training from scratch with VOC dataset doesn't work, as the dataset is too small. YOLO5 author recommends using COCO pre-trained and do transfer learning. Ahmed trained it in this manner previously.
- Ahmed provided VOC pre-trained YOLO5 small with old model definition
- Loaded the model successfully, can re-train the last layer on new model definition

**Comment by wernerchaodeeplite** [comment]
Successfully compressed YOLO5 small with 5% delta and --deepsearch. Result:
```
+--------------------------------------------------------------------------------------------------------------------------+
|                                                    Deeplite Profiler                                                     |
+-----------------------------------------+--------------------------+--------------------------+--------------------------+
|                              Param Name |               Enhancement|   Value (Optimized Model)|   Value (Reference Model)|
|                                         |                          |     Backend: TorchBackend|     Backend: TorchBackend|
+-----------------------------------------+--------------------------+--------------------------+--------------------------+
|                 Evaluation Metric (mAP) |                     -0.05|                    0.7863|                    0.8327|
|                         Model Size (MB) |                     2.80x|                    9.9588|                   27.8715|
|     Computational Complexity (GigaMACs) |                     1.62x|                    2.5652|                    4.1603|
|             Total Parameters (Millions) |                     2.80x|                    2.6106|                    7.3063|
|                   Memory Footprint (MB) |                     1.03x|                  294.8109|                  303.0680|
|                     Execution Time (ms) |                     1.07x|                    0.9582|                    1.0291|
+-----------------------------------------+--------------------------+--------------------------+--------------------------+
```

**Comment by wernerchaodeeplite** [comment]
new YOLO5 large achieved 0.898 mAP compared to current YOLO5 large model by Ahmed is 0.9 mAP. Will go ahead with optimizing with Neutrino.

**Comment by ehsab** [comment]
> new YOLO5 large achieved 0.898 mAP compared to current YOLO5 large model by Ahmed is 0.9 mAP. Will go ahead with optimizing with Neutrino.

Could you please upload the pretrained model on Amazon s3 and also update the zoo docs?

**Comment by wernerchaodeeplite** [comment]
> > new YOLO5 large achieved 0.898 mAP compared to current YOLO5 large model by Ahmed is 0.9 mAP. Will go ahead with optimizing with Neutrino.
> 
> Could you please upload the pretrained model on Amazon s3 and also update the zoo docs?

Yes I will.

**Comment by wernerchaodeeplite** [comment]
successfully optimized YOLO5 large with Neutrino, using --deepsearch and 0.01 delta
```
+--------------------------------------------------------------------------------------------------------------------------+
|                                                    Deeplite Profiler                                                     |
+-----------------------------------------+--------------------------+--------------------------+--------------------------+
|                              Param Name |               Enhancement|   Value (Optimized Model)|   Value (Reference Model)|
|                                         |                          |     Backend: TorchBackend|     Backend: TorchBackend|
+-----------------------------------------+--------------------------+--------------------------+--------------------------+
|                 Evaluation Metric (mAP) |                     -0.01|                    0.8880|                    0.8985|
|                         Model Size (MB) |                     2.39x|                   75.7119|                  181.1815|
|     Computational Complexity (GigaMACs) |                     1.45x|                   19.5910|                   28.4901|
|             Total Parameters (Millions) |                     2.39x|                   19.8474|                   47.4956|
|                   Memory Footprint (MB) |                     1.09x|                  850.2007|                  927.8117|
|                     Execution Time (ms) |                     1.14x|                    5.3550|                    6.1167|
+-----------------------------------------+--------------------------+--------------------------+--------------------------+```
